test/sql/index/art/constraints/test_art_concurrent_deletes.test

```diff
@@ -0,0 +1,137 @@
+# name: test/sql/index/art/constraints/test_art_concurrent_deletes.test
+# description: Test DELETE + INSERT in the main connection with concurrent deletes in other connections.
+# group: [constraints]
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+CREATE TABLE tbl (id INT PRIMARY KEY, payload VARCHAR);
+
+statement ok
+INSERT INTO tbl VALUES (1, 'first payload');
+
+# con1 open.
+
+statement ok con1
+BEGIN TRANSACTION;
+
+statement ok con1
+DELETE FROM tbl;
+
+statement ok con1
+INSERT INTO tbl VALUES (1, 'con1 payload');
+
+statement ok
+BEGIN TRANSACTION
+
+statement error
+DELETE FROM tbl;
+----
+<REGEX>:TransactionContext Error.*Conflict on tuple deletion.*
+
+statement ok
+COMMIT;
+
+# con1 and con2 open.
+
+statement ok con2
+BEGIN TRANSACTION;
+
+statement error con2
+DELETE FROM tbl;
+----
+<REGEX>:TransactionContext Error.*Conflict on tuple deletion.*
+
+statement ok
+BEGIN TRANSACTION;
+
+statement error
+DELETE FROM tbl;
+----
+<REGEX>:TransactionContext Error.*Conflict on tuple deletion.*
+
+statement ok
+COMMIT;
+
+statement ok con1
+COMMIT;
+
+statement ok con2
```

Comment:
Can we add a selection from the index on `con2` (`SELECT * FROM tbl WHERE id=1`), right after `con1` has committed (which should return the old value `first_payload`)?